contracts/governance/extensions/GovernorTimelockAccess.sol

```diff
@@ -223,12 +231,12 @@ abstract contract GovernorTimelockAccess is Governor {
         // If the proposal has been scheduled it will have an ETA and we have to externally cancel
         if (eta != 0) {
             for (uint256 i = 0; i < targets.length; ++i) {
-                (bool delayed, uint32 nonce) = _getManagerData(plan, i);
-                if (delayed) {
-                    // Attempt to cancel considering the operation could have been cancelled and rescheduled already
-                    uint32 canceledNonce = _manager.cancel(address(this), targets[i], calldatas[i]);
-                    if (canceledNonce != nonce) {
-                        revert GovernorMismatchedNonce(proposalId, nonce, canceledNonce);
+                (, bool withDelay, uint32 nonce) = _getManagerData(plan, i);
+                if (withDelay) {
+                    bytes32 operationId = keccak256(abi.encode(address(this), targets[i], calldatas[i]));
+                    if (nonce == _manager.getNonce(operationId)) {
+                        // Attempt to cancel considering the operation could have been cancelled and rescheduled already
+                        try _manager.cancel(address(this), targets[i], calldatas[i]) returns (uint32) {} catch {}
```

Comment:
Isn't the try-catch too "soft" here? The goal is to process exactly the case where it has already been cancelled, and not any possible reasons for the revert

Reply:
We want to make sure that the `cancel` function on the manager possibly failling for one of the calls does not prevent us moving forward (and attempting to cancel the other calls).

Do you have any idea for alternative implementations?

Reply:
It wouldn't be that ugly. What I want to understand is the benefit/drawbacks of doing that.

- What are the case where that would cause the cancel to fail?
- Are they cases where we want to fail?
- Are they cases where we dont want to fail?

Reply:
Added comments that should explain a bit more.